spec/models/user_spec.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+RSpec.describe User, type: :model do
+  before do
+    FactoryBot.create(:first_user)
+  end
+
+  context 'when user data is valid' do
+    let!(:user) { FactoryBot.build(:second_user) }
+
+    it 'should be valid' do
+      expect(user).to be_valid
+    end
+
+    it 'should have proper role' do
+      expect(user.role).to eq(FactoryBot.attributes_for(:second_user)[:role])
+    end
+
+    it 'should have proper gender' do
+      expect(user.gender).to eq(FactoryBot.attributes_for(:second_user)[:gender])
+    end
+
+    it 'should have proper status' do
+      expect(user.status).to eq('pending')
```

Comment:
here you are checking factoryBot instance is created successfully or not. This is not a proper way of writing rspec of model. read this
[shoulda-matchers]( https://github.com/thoughtbot/shoulda-matchers)

Reply:
@kth-wtag apu, when writing spec for a Model, using `shoulda-matcher` I have checked whether enum is defined or not. However, when creating an instance of that Model, should not I check whether that specific instance have correct value for the defined `enum` or not as well?